File: leetcode.com/0860_lemonade_change/solution.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        five = ten = 0

        for bill in bills:
            if bill == 5:
                five += 1
                continue

            if bill == 10:
                if five == 0:
                    return False

                five -= 1
                ten += 1
                continue

            if bill == 20:
                if ten > 0 and five > 0:
                    ten -= 1
                    five -= 1
                    continue

                elif five > 3:
                    five -= 3
                    continue

                return False

        return True

File: leetcode.com/0860_lemonade_change/solution.c
bool lemonadeChange(int *bills, int billsSize)
{
    int total_five = 0;
    int total_ten = 0;

    for (int i = 0 ; i < billsSize ; i++) {
        switch (bills[i]) {
        case 5:
            total_five += 1;
            continue;

        case 10:
            if (total_five == 0)
                return false;

            total_ten += 1;
            total_five -= 1;
            continue;

        case 20:
            if (total_ten > 0 && total_five > 0) {
                total_ten -= 1;
                total_five -= 1;
                continue;
            }

            if (total_five < 3)
                return false;
   
            total_five -= 3;
            continue;
        }
    }

    return true;
}
